src/sinks/azure_blob/config.rs

```diff
@@ -123,11 +134,26 @@ impl AzureBlobSinkConfig {
             .blob_append_uuid
             .unwrap_or(DEFAULT_FILENAME_APPEND_UUID);
 
+        let transformer = self.encoding.transformer();
+        let (framer, serializer) = self.encoding.clone().encoding();
+        let framer = match (framer, &serializer) {
+            (Some(framer), _) => framer,
+            (None, Serializer::Json(_)) => CharacterDelimitedEncoder::new(b',').into(),
+            (None, Serializer::Native(_)) => {
+                // TODO: We probably want to use something like octet framing here.
+                return Err("Native encoding is not implemented for this sink yet".into());
+            }
+            (None, Serializer::NativeJson(_) | Serializer::RawMessage(_)) => {
+                NewlineDelimitedEncoder::new().into()
+            }
```

Comment:
@lukesteensen curious about which framing choices you would take for the `Native`/`NativeJson` serializer

Reply:
In #12407 I used length-delimited framing as the default for `Native`. `NativeJson` can just be the same as JSON or NDJSON. Eventually we should do a sink-equivalent of #12407 so that this logic doesn't need to live in each sink.

Reply:
👍 to octet framing as the default for `Native` to match `decoding`. Agreed that it'd be nice to refactor this like https://github.com/vectordotdev/vector/pull/12407 .